blobs/howto/dotnet/dotnet-v12/CRUD_DataLake.cs

```diff
@@ -222,55 +239,61 @@ public async Task ListFilesInDirectory(DataLakeFileSystemClient fileSystemClient
-        // Upload files to the directory.
+        // Upload a file to the directory
         //----------------------------------------------------------
 
         // <Snippet_UploadFile>
-        public async Task UploadFile(DataLakeFileSystemClient fileSystemClient)
+        public async Task UploadFile(
+            DataLakeFileSystemClient fileSystemClient,
+            string directoryName,
+            string fileName,
+            string localPath)
         {
             DataLakeDirectoryClient directoryClient =
-                fileSystemClient.GetDirectoryClient("my-directory");
-
-            DataLakeFileClient fileClient = await directoryClient.CreateFileAsync("uploaded-file.txt");
-
-            FileStream fileStream =
-                File.OpenRead("C:\\Users\\contoso\\Temp\\file-to-upload.txt");
-
-            long fileSize = fileStream.Length;
+                fileSystemClient.GetDirectoryClient(directoryName);
 
-            await fileClient.AppendAsync(fileStream, offset: 0);
+            DataLakeFileClient fileClient = 
+                directoryClient.GetFileClient(fileName);
 
-            await fileClient.FlushAsync(position: fileSize);
+            FileStream fileStream = File.OpenRead(localPath);
 
+            await fileClient.UploadAsync(content: fileStream, overwrite: true);
         }
         // </Snippet_UploadFile>
 
         #endregion
 
-        #region Upload files to a directory in bulk
+        #region Append data to a file
 
         // ---------------------------------------------------------
-        // Upload files to the directory - bulk uploads
+        // Append data to a file
         //----------------------------------------------------------
 
-        // <Snippet_UploadFileBulk>
-        public async Task UploadFileBulk(DataLakeFileSystemClient fileSystemClient)
+        // <Snippet_AppendDataToFile>
+        public async Task AppendDataToFile(
+            DataLakeFileSystemClient fileSystemClient,
+            string directoryName,
+            string fileName,
+            string localPath)
         {
             DataLakeDirectoryClient directoryClient =
-                fileSystemClient.GetDirectoryClient("my-directory");
+                fileSystemClient.GetDirectoryClient(directoryName);
 
-            DataLakeFileClient fileClient = directoryClient.GetFileClient("uploaded-file.txt");
+            DataLakeFileClient fileClient = 
+                directoryClient.GetFileClient(fileName);
 
-            FileStream fileStream =
-                File.OpenRead("C:\\Users\\contoso\\file-to-upload.txt");
+            long fileSize = fileClient.GetProperties().Value.ContentLength;
 
-            await fileClient.UploadAsync(fileStream);
+            FileStream fileStream = File.OpenRead(localPath);
 
+            await fileClient.AppendAsync(fileStream, offset: fileSize);
+
+            await fileClient.FlushAsync(position: fileSize + fileStream.Length);
```

Comment:
Do we not think it's a little confusing for someone adapting this sample to see someone appending from the start of a local file to the end of an existing datalake file? The start of an existing file on disk is probably the start for a reason.

Perhaps this snippet can take an arbitrary `Stream` as an argument instead of `localPath`.

Reply:
Makes sense, thanks.